File: src/test/resources/js/RFunctions.js
class RFunctions {
    factorialRecursive(n) {
        if (n <= 1) {
            return 1;
        } else {
            return n * this.factorialRecursive(n - 1);
        }
    }

    findMaxIterative(elements) {
        let max = elements[0];
        for (let element of elements) {
            if (element > max) {
                max = element;
            }
        }
        return max;
    }
}

File: src/test/resources/java/RFunctions.java
import java.util.List;

public class RFunctions {

    public int factorialRecursive(int n) {
        if (n <= 1) {
            return 1;
        } else {
            return n * factorialRecursive(n - 1);
        }
    }

    public int findMaxIterative(List<Integer> elements) {
        int max = elements.get(0);
        for (int element : elements) {
            if (element > max) {
                max = element;
            }
        }
        return max;
    }
}
